orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/strategies/RollingRedBlackStrategy.groovy

```diff
@@ -1,4 +1,136 @@
 package com.netflix.spinnaker.orca.clouddriver.pipeline.servergroup.strategies
 
-class RollingRedBlackStrategy {
+import com.netflix.spinnaker.orca.clouddriver.pipeline.cluster.DisableClusterStage
+import com.netflix.spinnaker.orca.clouddriver.pipeline.servergroup.ResizeServerGroupStage
+import com.netflix.spinnaker.orca.clouddriver.pipeline.servergroup.support.DetermineTargetServerGroupStage
+import com.netflix.spinnaker.orca.clouddriver.pipeline.servergroup.support.TargetServerGroup
+import com.netflix.spinnaker.orca.kato.pipeline.support.SourceResolver
+import com.netflix.spinnaker.orca.pipeline.model.Execution
+import com.netflix.spinnaker.orca.pipeline.model.Stage
+import com.netflix.spinnaker.orca.pipeline.model.SyntheticStageOwner
+import groovy.util.logging.Slf4j
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.context.ApplicationContext
+import org.springframework.context.ApplicationContextAware
+import org.springframework.stereotype.Component
+
+import static com.netflix.spinnaker.orca.pipeline.StageDefinitionBuilder.StageDefinitionBuilderSupport.newStage
+
+@Slf4j
+@Component
+class RollingRedBlackStrategy implements Strategy, ApplicationContextAware {
+  final String name = "rollingredblack"
+
+  @Autowired
+  DisableClusterStage disableClusterStage
+
+  @Autowired
+  ResizeServerGroupStage resizeServerGroupStage
+
+  @Autowired
+  DetermineTargetServerGroupStage determineTargetServerGroupStage
+
+  @Autowired
+  SourceResolver sourceResolver
+
+  @Override
+  <T extends Execution<T>> List<Stage<T>> composeFlow(Stage<T> stage) {
+    def stages = []
+    def stageData = stage.mapTo(RollingRedBlackStageData)
+    def cleanupConfig = AbstractDeployStrategyStage.CleanupConfig.fromStage(stage)
+
+    Map baseContext = [
+        (cleanupConfig.location.singularType()): cleanupConfig.location.value,
+        cluster                                : cleanupConfig.cluster,
+        credentials                            : cleanupConfig.account,
+        cloudProvider                          : cleanupConfig.cloudProvider,
+    ]
+
+    Map savedCapacity = stageData.savedCapacity ?: stageData.capacity
+    if (!savedCapacity) {
```

Comment:
Ah right, @robfletcher - I added this because this code path seems be called twice when building the stages, do you know why that is?